allauth/account/adapter.py

```diff
@@ -296,6 +298,11 @@ def clean_password(self, password, user=None):
         validate_password(password, user)
         return password
 
+    def validate_unique_email(self, email):
+        if email and email_address_exists(email):
```

Comment:
Can we move the `if email` part back into the original place (forms) -- so that people overriding adapters don't have to be bothered with empty values?